packages/transcript-view/src/transcript-view.ts

```diff
@@ -0,0 +1,383 @@
+import {
+  LitElement,
+  html,
+  css,
+  customElement,
+  property,
+  PropertyValues,
+  TemplateResult,
+  CSSResult,
+} from 'lit-element';
+import './transcript-entry';
+import './duration-formatter';
+import TranscriptEntryConfig from './models/transcript-entry-config';
+
+@customElement('transcript-view')
+export default class TranscriptView extends LitElement {
+  @property({ type: [TranscriptEntryConfig] }) entries: TranscriptEntryConfig[] = [];
+
+  @property({ type: Number }) currentTime = 0;
+
+  @property({ type: Number }) topContextHeight = 50;
+
+  @property({ type: Number }) bottomContextHeight = 50;
+
+  @property({ type: Boolean }) autoScroll = true;
+
+  @property({ type: Number }) selectedSearchResultIndex = 0;
+
+  @property({ type: Boolean }) showContextZones = false;
+
+  @property({ type: Number }) private timeScrollTop = 0;
+
+  @property({ type: TranscriptEntryConfig }) private currentEntry:
+    | TranscriptEntryConfig
+    | undefined;
+
+  private scrollTimerDelay = 15000;
+
+  private scrollResumeTimerId = -1;
+
+  render(): TemplateResult {
+    return html`
+      <div class="container">
+        ${this.showContextZones ? this.contextZoneDevTemplates : ''}
+
+        <div class="scroll-container" id="scroll-container" @wheel=${this.didScroll}>
+          <div class="col time">
+            ${this.timeDisplayTemplate}
+          </div>
+
+          <div class="col">
+            ${this.autoScrollButtonTemplate}
+            ${(this.entries ? this.entries : []).map((entry: TranscriptEntryConfig) =>
+              this.transcriptEntryTemplate(entry),
+            )}
+          </div>
+        </div>
+      </div>
+    `;
+  }
+
+  private get autoScrollButtonTemplate(): TemplateResult {
+    return html`
+      <button
+        @click=${this.enableAutoScroll}
+        class="auto-scroll-button"
+        style="display: ${this.autoScroll ? 'none' : 'inline-block'}"
```

Comment:
**Suggestion**: replace the style property toggle with a class toggle. This will make it easier to change which property(ies) are toggled based on boolean state and make it easier to test in unit test libraries that do not fully render the DOM.